ios/RNMobileDeviceManager.m

```diff
@@ -0,0 +1,94 @@
+#import "RNMobileDeviceManager.h"
+
+// For RCTMakeError
+#if __has_include(<React/RCTUtils.h>)
+#import <React/RCTUtils.h>
+#elif __has_include("RCTUtils.h")
+#import "RCTUtils.h"
+#else
+#import "React/RCTUtils.h"
+#endif
+
+// Used to send events to JS
+#if __has_include(<React/RCTBridge.h>)
+#import <React/RCTBridge.h>
+#elif __has_include("RCTBridge.h")
+#import "RCTBridge.h"
+#else
+#import "React/RCTBridge.h"
+#endif
+
+#if __has_include(<React/RCTEventDispatcher.h>)
+#import <React/RCTEventDispatcher.h>
+#elif __has_include("RCTEventDispatcher.h")
+#import "RCTEventDispatcher.h"
+#else
+#import "React/RCTEventDispatcher.h"
+#endif
+
+@implementation MobileDeviceManager
+
+@synthesize bridge = _bridge;
+
+- (instancetype)init
+{
+    if ((self = [super init])) {
+        // Add Notification Center observer to be alerted of any change to NSUserDefaults.
+        // Managed app configuration changes pushed down from an MDM server appear in NSUSerDefaults.
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(userDefaultsDidChange:) name:@"NSUserDefaultsDidChangeNotification" object:self];
```

Comment:
> I don't think this will observe configuration changes when the application isn't running, something to be aware of

Do we want it to observe config changes when the app isn't running?

Reply:
> This is going to fire when any defaults change. We may want to take a look at using a KVO implementation on NSUserDefaults

Tried implementing a KVO for `NSUserDefaults` but it didn't work. Not too familiar with KVOs but after spending some time looking around, seems like it's not actually KVO compliant and only works when `defaults are changed from inside the app itself, not from the settings bundle.` Also, Apple's official guide on Managed App Config uses `NSNotificationCenter` as well.

https://stackoverflow.com/questions/10871860/nsuserdefaultsdidchangenotification-whats-the-name-of-the-key-that-changed